### ISSUE TITLE
Improved handling of content keys during offer/accept flow

### ISSUE THREAD

**Issue by njgheorghita** [ethereum/trin#679]
A couple possible improvements to our offer / accept flow....

- Currently, a single trin client will accept multiple offers for the same content key simultaneously. Only after the utp tx is complete and the content is available in our database do we start to refuse offers for a stored content key. This could be somewhat significant since utp transfers for bodies / receipts are somewhat lengthy (30-60 seconds). During this window it's certainly likely that a trin node will initiate multiple new utp txs with other clients for the exact same content.

- Currently, if a utp tx fails, trin simply drops the entire offer/accept flow. Especially if we implement the above suggestion, it would be beneficial to introduce some sort of `try_again` functionality for dropped utp transfers. This might simply consist of initiating a `RFC` request for the dropped content key.

**Comment by njgheorghita** [comment]
closing in favor of #856
